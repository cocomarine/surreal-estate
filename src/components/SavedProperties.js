import axios from "axios";
import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import SavedPropertiesCard from "./SavedPropertyCard";
import Alert from "./Alert";
import "../styles/saved-properties.css";

const SavedProperties = ({ userID }) => {
  if (!userID)
    return (
      <div className="login-msg">Please login to view saved properties.</div>
    );

  const initialState = {
    savedProperties: [],
  };

  const [savedProperties, setSavedProperties] = useState(
    initialState.savedProperties
  );
  const [alert, setAlert] = useState({ message: "" });

  useEffect(() => {
    axios
      .get("http://localhost:4000/api/v1/Favourite?populate=propertyListing")
      .then((res) => {
        setSavedProperties(res.data);
        setAlert({
          message: "",
        });
      })
      .catch(() => {
        setAlert({
          message: "Server error. Please try again later.",
        });
      });
  }, []);

  const handleUnsaveProperty = (propertyId) => {
    axios
      .delete(`http://localhost:4000/api/v1/Favourite/${propertyId}`)
      .then(() => {
        axios
          .get(
            "http://localhost:4000/api/v1/Favourite?populate=propertyListing"
          )
          .then((res) => {
            setSavedProperties(res.data);
          });
      })
      .catch((err) => console.error(err));
  };

  return (
    <div className="saved-properties-container">
      <Alert message={alert.message} success={alert.isSuccess} />
      {savedProperties.map((savedProperty) => (
        <div className="item" key={savedProperty._id}>
          <SavedPropertiesCard
            userID={savedProperties.userId}
            propertyId={savedProperty._id}
            title={savedProperty.propertyListing.title}
            city={savedProperty.propertyListing.city}
            onUnsaveProperty={handleUnsaveProperty}
          />
        </div>
      ))}
    </div>
  );
};

SavedProperties.propTypes = {
  userID: PropTypes.string.isRequired,
};

export default SavedProperties;
