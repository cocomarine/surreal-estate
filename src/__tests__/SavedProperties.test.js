import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import SavedProperties from "../components/SavedProperties";

describe("SavedProperties", () => {
  let testUserID = "test userID";

  it("renders correctly", () => {
    const rendered = renderer.create(<SavedProperties userID={testUserID} />);
    expect(rendered).toMatchSnapshot();
  });

  it("returns login message when userID not truthy", () => {
    testUserID = "";
    render(<SavedProperties userID={testUserID} />);
    const text = screen.getByText("Please login to view saved properties.");

    expect(text).toBeInTheDocument();
  });
});
