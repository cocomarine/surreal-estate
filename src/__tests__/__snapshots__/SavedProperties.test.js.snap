// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedProperties renders correctly 1`] = `
<div
  className="saved-properties-container"
/>
`;
